deployment/scripts/kind/install.sh

```diff
@@ -0,0 +1,754 @@
+        set -e
+        
+        if [ $config_result -eq 0 ]; then
+            log_success "LLMInferenceService configured"
+            return 0
+        else
+            retry_count=$((retry_count + 1))
+            if [ $retry_count -lt $max_retries ]; then
+                log_warn "Webhook not ready yet, waiting 15 seconds before retry..."
+                sleep 15
+            else
+                log_error "Failed to configure LLMInferenceService after $max_retries attempts"
+                return 1
+            fi
+        fi
+    done
+}
+
+# Build local MaaS API image with K8s-first code
+build_maas_api_image() {
+    log_info "Building local maas-api image with K8s-first code..."
+
+    cd "$PROJECT_ROOT/maas-api"
+    
+    # Determine container engine and build image
+    if command_exists docker && docker ps >/dev/null 2>&1; then
+        log_info "Building MaaS API image with Docker..."
+        make build-image CONTAINER_ENGINE=docker REPO=localhost/maas-api TAG=dev >/dev/null 2>&1 || {
+            log_error "Failed to build maas-api image with Docker"
+            return 1
+        }
+    elif command_exists podman && podman ps >/dev/null 2>&1; then
+        log_info "Building MaaS API image with Podman..."
+        make build-image CONTAINER_ENGINE=podman REPO=localhost/maas-api TAG=dev >/dev/null 2>&1 || {
+            log_error "Failed to build maas-api image with Podman"
+            return 1
+        }
+    else
+        log_error "No working container engine found"
+        return 1
+    fi
+
+    log_info "Loading maas-api image into Kind cluster..."
+    kind load docker-image localhost/maas-api:dev --name "$CLUSTER_NAME" >/dev/null 2>&1 || {
+        log_error "Failed to load image into Kind"
+        return 1
+    }
+
+    cd "$PROJECT_ROOT"
+    log_success "MaaS API image built and loaded"
+}
+
+# Deploy MaaS components
+deploy_maas() {
+    log_info "Deploying MaaS components..."
+
+    # Build local maas-api image first
+    build_maas_api_image
+
+    # Apply Kustomize overlay
+    log_info "Applying MaaS configuration..."
+    kubectl apply -k "$PROJECT_ROOT/deployment/overlays/kind/" >/dev/null 2>&1 || {
+        log_warn "Some warnings during MaaS deployment (checking for details...)"
+        kubectl apply -k "$PROJECT_ROOT/deployment/overlays/kind/" 2>&1 | grep -E "(Error|Failed)" || true
+    }
+
+    # Wait for Gateway to be created and service to be provisioned
+    log_info "Waiting for Gateway to be provisioned..."
+    sleep 5
+    kubectl wait --for=condition=Programmed gateway/maas-gateway -n istio-system --timeout=300s >/dev/null 2>&1 || true
+
+    # Patch the Gateway service to use fixed NodePorts (30080, 30443)
+    # This is required for Kind's extraPortMappings to work
+    log_info "Configuring Gateway service NodePorts..."
+    kubectl patch svc maas-gateway-istio -n istio-system --type='json' -p='[
+      {"op":"replace","path":"/spec/ports/1/nodePort","value":30080},
```

Comment:
The script hardcodes NodePort values (30080, 30443) without validating they match the `kind-config.yaml` file. If someone changes the config file, the script will fail silently or patch incorrectly. I would recommend to add validation to extract NodePort values from `KIND_CONFIG` or verify they match before patching.